int32 allied_type
---
int32 height
int32 width
int32 error